From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Finkel <me@martinfinkel.com>
Date: Mon, 1 Jun 2026 14:00:00 +0200
Subject: [PATCH] iconv: drop UWP GetHandleInformation polyfill

llvm-mingw now declares GetHandleInformation in <handleapi.h> and
defines HANDLE_FLAG_INHERIT in <winbase.h> for UWP, so the iconv
polyfill collides with the system declaration and breaks the build.
Stop applying the patch since the polyfill is no longer needed.
---
 contrib/src/iconv/rules.mak | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/contrib/src/iconv/rules.mak b/contrib/src/iconv/rules.mak
--- a/contrib/src/iconv/rules.mak
+++ b/contrib/src/iconv/rules.mak
@@ -27,10 +27,6 @@ iconv: libiconv-$(LIBICONV_VERSION).tar.gz .sum-iconv
 	# use CreateFile2 instead of CreateFile in UWP
 	$(APPLY) $(SRC)/iconv/0001-Use-CreateFile2-in-UWP-builds.patch
 
-	# fix forbidden UWP call which can't be upstreamed as they won't
-	# differentiate for winstore, only _WIN32_WINNT
-	$(APPLY) $(SRC)/iconv/0001-do-not-call-GetHandleInformation-in-Winstore-apps.patch
-
 	$(UPDATE_AUTOCONFIG) && cd $(UNPACK_DIR) && mv config.guess config.sub build-aux
 	$(UPDATE_AUTOCONFIG) && cd $(UNPACK_DIR) && mv config.guess config.sub libcharset/build-aux
 	$(MOVE)
-- 
2.40.0
